flytekit/deck/deck.py

```diff
@@ -174,3 +174,66 @@ def get_deck_template() -> "Template":
         autoescape=select_autoescape(enabled_extensions=("html",)),
     )
     return env.get_template("template.html")
+
+
+class PythonDependencyDeck(Deck):
+    """
+    PythonDependencyDeck is a deck that contains information about the library name and version.
+    """
+
+    def __init__(self, name: Optional[str] = "Python Dependencies", html: Optional[str] = ""):
+        super().__init__(name, html)
+
+    @property
+    def html(self) -> str:
+        import json
+        import subprocess
+        import sys
+
+        import pandas as pd
+        from flytekitplugins.deck.renderer import TableRenderer
```

Comment:
Can we use `flytekit.deck.MarkdownRenderer` instead? This would make it possible to not require `flytekitplugins-deck-standard` to get the dependency information. `flytekitplugins-deck-standard` is a pretty heavy package because of all it's dependencies.

Reply:
I use `MarkdownRenderer` instead of `TableRenderer`
https://github.com/flyteorg/flytekit/blob/0091f25e8bceb48513271a2e5266535114061485/flytekit/deck/deck.py#L207

Deck:
<img width="1637" alt="Screenshot 2024-03-25 at 6 05 34 PM" src="https://github.com/flyteorg/flytekit/assets/14800485/4eff1027-48d0-4347-bcb7-ebe62b55d285">

when the version is too long, the layout becomes confusing

Reply:
if we use the `MarkdownRenderer` we lose the control over the overall html (so this idea of adding a button that copies the contents of a hidden div won't work). That said, the `TableRenderer` only [depends on pandas](https://github.com/flyteorg/flytekit/blob/d23eb440a9cdb730e3802d7fa93b5aafd0670779/plugins/flytekit-deck-standard/flytekitplugins/deck/renderer.py#L135-L177) and could easily be moved to core.